org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java

```diff
@@ -827,30 +827,38 @@ public Object invoke(Object proxy, Method method, Object[] args)
 					if (cls.isAnnotation()) {
 						val = method.getDefaultValue();
 					} else if (method.isDefault()) {
-						double javaVersion = Double.parseDouble(
-								System.getProperty("java.class.version"));
-						double java8 = 52.0d;
-						if (javaVersion > java8) {
-							val = MethodHandles.lookup()
-									.findSpecial(method.getDeclaringClass(),
-											method.getName(),
-											MethodType.methodType(
-													method.getReturnType(),
-													new Class[] {}),
-											method.getDeclaringClass())
-									.bindTo(proxy)
-									.invokeWithArguments(args);
-						} else {
-							Constructor<Lookup> constructor = Lookup.class
-									.getDeclaredConstructor(Class.class);
-							if (!constructor.isAccessible()) {
-								constructor.setAccessible(true);
+						try {
+							double javaVersion = Double.parseDouble(
+									System.getProperty("java.class.version"));
+							double java8 = 52.0d;
+							if (javaVersion > java8) {
+								val = MethodHandles.lookup()
+										.findSpecial(method.getDeclaringClass(),
+												method.getName(),
+												MethodType.methodType(
+														method.getReturnType(),
+														new Class[] {}),
+												method.getDeclaringClass())
+										.bindTo(proxy)
+										.invokeWithArguments(args);
+							} else {
+								Constructor<Lookup> constructor = Lookup.class
+										.getDeclaredConstructor(Class.class);
+								if (!constructor.isAccessible()) {
+									constructor.setAccessible(true);
+								}
+								val = constructor.newInstance(cls)
+										.in(cls)
+										.unreflectSpecial(method, cls)
+										.bindTo(proxy)
+										.invokeWithArguments(args);
 							}
-							val = constructor.newInstance(cls)
-									.in(cls)
-									.unreflectSpecial(method, cls)
-									.bindTo(proxy)
-									.invokeWithArguments(args);
+							return val;
```

Comment:
I think that returning here is too early. You miss the final conversion at the end which may be necessary to convert the method's return type into the conversion's desired type.

Reply:
I thought that we are in the block where we execute the default method. There should be guarantied, by the compiler that the returned value is the conversion's desired type.
Could you help me to understand better why it is needed there?
 
It is there, because the default method returning null (as wished).
I will not go into the block, because it will lead to an exception

```
if (val == null) {
	if (args != null && args.length == 1) {
		val = args[0];
	} else {
		throw new ConversionException(
			"No value for property: " + propName);
	}
}
```
I could change it to just return if the default method returns null, but not sure.